### ISSUE TITLE
Wrong error for for..in statement

### ISSUE THREAD

**Issue by JsonFreeman** [microsoft/TypeScript#154]
``` TS
var idx = 0;
var m = [1, 2, 3, 4, 5];
var n = [[]] || [];
for (n[idx++] in m);
```

We give the error: "Variable declarations of a 'for' statement must be of types 'string' or 'any'."

But `n[idx++]` is not a variable declaration!

It should say: "Left hand side of a 'for' statement must be of types 'string' or 'any'."
